### PR TITLE
Create script to generate DOIs

### PR DESCRIPTION
**Pull request by lfarrell**
https://jira.lib.unc.edu/browse/HYC-776
Depends on: https://gitlab.lib.unc.edu/cdr/vagrant-rails/merge_requests/76

### CODE REVIEW

File: app/helpers/hyc/doi_create.rb [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        work = ActiveFedora::Base.find(record['id'])
+        work.doi = "https://doi.org/#{doi}"
+        work.save!
+        Rails.logger.info "DOI created for record #{record['id']} via DataCite."
+      else
+        Rails.logger.warn "Unable to create DOI for record #{record['id']} via DataCite. DOI not added. Reason: \"#{response}\""
+      end
+    end
+
+    def create_single_doi(record_id)
```
[the remark this answers is not in the document]
Comment (lfarrell) [reply app/helpers/hyc/doi_create.rb]: Yeah, it was also pulling in fileSets

[the remark this answers is not in the document]
Comment (lfarrell) [reply app/helpers/hyc/doi_create.rb]: Didn't know if we wanted to create one off DOIs, but I guess we'd just use the web interface for that.

File: app/helpers/hyc/doi_create.rb [hunk 2]
```diff
@@ -124,6 +124,7 @@ def format_data(record)
     end
 
     def create_doi(record)
+      puts "Creating DOI for #{record['id']}"
```
Comment (bbpennel) [inline app/helpers/hyc/doi_create.rb]: Seems like you would want to use Rails.logger like you are in the rest of this script?

File: app/helpers/hyc/doi_create.rb [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+      end
+
+      sizes = parse_field(record, 'extent_tesim')
+      unless sizes.blank?
+        data[:data][:attributes][:sizes] = sizes
+      end
+
+      subjects = parse_field(record, 'subject_tesim')
+      unless subjects.blank?
+        data[:data][:attributes][:sizes] = subjects
```
Comment (bbpennel) [inline app/helpers/hyc/doi_create.rb]: subjects are being put into sizes, should be a different field

**Pull request merged**